### PR TITLE
Add examples for Compose

### PR DESCRIPTION
**Pull request by yevhenii-nadtochii**
This PR adds the basic examples of using JxBrowser with Compose Desktop.

The following snippets have been created:

1. `ComposeBrowserView`.
2. `BitmapToComposeImage`.
3. `CustomContextMenu`.
4. `DefaultOpenPopupCallback`.

Please note, JavaFX is a dependency now due to migration to Java 17.

### CODE REVIEW

**Comment by yevhenii-nadtochii** [comment]
@armiol @Danil-Didkovskiy PTAL

File: .idea/codeStyles/Project.xml [hunk 1]
```diff
@@ -22,6 +22,7 @@
     <option name="DOWHILE_BRACE_FORCE" value="3" />
     <option name="WHILE_BRACE_FORCE" value="3" />
     <option name="FOR_BRACE_FORCE" value="3" />
+    <option name="SOFT_MARGINS" value="80" />
```
Comment (armiol) [inline .idea/codeStyles/Project.xml:25]: I would discuss this with the core JxBrowser team as well.

File: build.gradle.kts [hunk 2]
```diff
@@ -88,16 +99,20 @@ subprojects {
...
         implementation(jxbrowser.swt)
 
-        // Dependency on a SWT for the current platform.
+        // Dependency on an SWT for the current platform.
         implementation(Swt.toolkitDependency)
 
         // Depend on Guava for the Resources utility class used for loading resource files into strings.
         implementation("com.google.guava:guava:$guavaVersion")
-
-        implementation(files("$rootDir/examples/src/main/resources/resource.jar"))
```
Comment (armiol) [inline build.gradle.kts:100]: What was this?

File: examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/BitmapToComposeImage.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import com.teamdev.jxbrowser.compose.graphics.toBufferedImage
+import com.teamdev.jxbrowser.dsl.Engine
+import com.teamdev.jxbrowser.dsl.browser.navigation
+import com.teamdev.jxbrowser.engine.RenderingMode
+import java.io.File
+import javax.imageio.ImageIO
+
+/**
+ * This example demonstrates how to take a bitmap of the loaded web page,
+ * convert it to a Java AWT image and save to a PNG file.
```
Comment (armiol) [inline examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/BitmapToComposeImage.kt:32]: "... , and save ..."

File: examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/ComposeBrowserView.kt [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+import androidx.compose.runtime.remember
+import androidx.compose.ui.window.singleWindowApplication
+import com.teamdev.jxbrowser.compose.BrowserView
+import com.teamdev.jxbrowser.dsl.Engine
+import com.teamdev.jxbrowser.dsl.browser.navigation
+import com.teamdev.jxbrowser.engine.RenderingMode
+
+/**
+ * This example demonstrates how to embed Compose [BrowserView]
+ * into Compose application.
```
Comment (armiol) [inline examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/ComposeBrowserView.kt:33]: "... into a Compose application".

File: examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/CustomContextMenu.kt [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+            browser.navigation.reload()
+            menuShowed = false
+        }) {
+            Text("Reload")
+        }
+    }
+
+    DisposableEffect(Unit) {
+
+        // Register a callback, which shows the menu when a user right-clicks
```
Comment (armiol) [inline examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/CustomContextMenu.kt:98]: I would rephrase (and re-layout). E.g.

```
// Register a callback to display the menu
// upon right-clicking on a web page.
```

File: examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/popup/DefaultOpenPopupCallback.kt [hunk 6]
```diff
@@ -0,0 +1,110 @@
...
+ * for Compose UI toolkit.
+ *
+ * It creates and shows a new window with the embedded pop-up browser
+ * each time [OpenPopupCallback] is invoked.
+ *
+ * This implementation matches the one provided in JxBrowser by default.
+ * Custom implementation of [OpenPopupCallback] takes precedence
+ * over the default one.
+ *
+ * This example snippet is meant to be used as a prototype for clients
```
Comment (armiol) [inline examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/popup/DefaultOpenPopupCallback.kt:47]: Do we need this paragraph at all? Why we don't have it in all other examples?

I say we kill it.

File: examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/popup/DefaultOpenPopupCallback.kt [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+    val scope = rememberCoroutineScope()
+
+    // Add browser view.
+    BrowserView(browser)
+
+    // Add pop-up windows, if any.
+    for (popup in popups) {
+        // We associate each `PopupWindow` with a unique key. This helps Compose
+        // efficiently manage the lifecycle of popup components when the list
+        // of popups changes.
```
Comment (armiol) [inline examples/src/main/kotlin/com/teamdev/jxbrowser/examples/compose/popup/DefaultOpenPopupCallback.kt:68]: Please make sure to stay with "pop-up", not with "popup", throughout the texts.

Please review the official docs. And I suggest we use the same wording.

Right now, in this changeset, usage of this word is not too consistent.

**Review by armiol — CHANGES_REQUESTED** [review]
@yevhenii-nadtochii please see my comments.

File: .idea/codeStyles/Project.xml (hunk 1 above)
Comment (yevhenii-nadtochii) [reply .idea/codeStyles/Project.xml]: Discussed with @vladimir-ikryanov. 

Decided to set `80` for the hard wrapping, so that snippets were not long and fit well on the site.

File: build.gradle.kts (hunk 2 above)
Comment (yevhenii-nadtochii) [reply build.gradle.kts:100]: The file is used by one of the snippets, but it is not present anymore. Gradle fails such a build now.

I've created the issue: https://github.com/TeamDev-IP/JxBrowser-Docs/issues/921

**Comment by yevhenii-nadtochii** [comment]
@armiol PTAL

**Review by armiol — APPROVED** [review]
